hack/release/common.sh

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+RELEASE_ACR=${RELEASE_ACR:-ksnap.azurecr.io} # will always be overridden
+RELEASE_REPO_ACR=${RELEASE_REPO_ACR:-${RELEASE_ACR}/public/aks/karpenter}
+RELEASE_REPO_MAR=mcr.microsoft.com/aks/karpenter
+SNAPSHOT_ACR=${SNAPSHOT_ACR:-ksnap.azurecr.io}
+SNAPSHOT_REPO_ACR=${SNAPSHOT_REPO_ACR:-${SNAPSHOT_ACR}/karpenter/snapshot}
+
+CURRENT_MAJOR_VERSION="0"
+
+snapshot() {
+  local commit_sha version helm_chart_version
+
+  commit_sha="${1}"
+  version="${commit_sha}"
+  helm_chart_version="${CURRENT_MAJOR_VERSION}-${commit_sha}"
+
+  echo "Release Type: snapshot
+Release Version: ${version}
+Commit: ${commit_sha}
+Helm Chart Version ${helm_chart_version}"
+
+  authenticate "${SNAPSHOT_ACR}"
+  buildAndPublish "${SNAPSHOT_REPO_ACR}" "${version}" "${helm_chart_version}" "${commit_sha}"
+}
+
+release() {
+  local commit_sha version helm_chart_version
+
+  commit_sha="${1}"
+  version="${2}"
+  helm_chart_version="${version}"
+
+  echo "Release Type: stable
+Release Version: ${version}
+Commit: ${commit_sha}
+Helm Chart Version ${helm_chart_version}"
+
+  authenticate "${RELEASE_ACR}"
+  buildAndPublish "${RELEASE_REPO_ACR}" "${version}" "${helm_chart_version}" "${commit_sha}" \
+    "${RELEASE_REPO_MAR}" # repo override for Helm chart
+}
+
+authenticate() {
+  local acr
+
+  acr="$1"
+  az acr login -n "${acr}"
+}
+
+buildAndPublish() {
+  local oci_repo version helm_chart_version commit_sha date_epoch build_date img img_repo img_tag img_digest
+
+  oci_repo="${1}"
+  version="${2}"
+  helm_chart_version="${3}"
+  commit_sha="${4}"
+
+  date_epoch="$(dateEpoch)"
+  build_date="$(buildDate "${date_epoch}")"
+
+  img="$(GOFLAGS=${GOFLAGS:-} \
+    SOURCE_DATE_EPOCH="${date_epoch}" KO_DATA_DATE_EPOCH="${date_epoch}" KO_DOCKER_REPO="${oci_repo}" \
+    ko publish -B --sbom none -t "${version}"     ./cmd/controller)"
+  img_nap="$(GOFLAGS="${GOFLAGS:-} -tags=ccp" \
+    SOURCE_DATE_EPOCH="${date_epoch}" KO_DATA_DATE_EPOCH="${date_epoch}" KO_DOCKER_REPO="${oci_repo}" \
+    ko publish -B --sbom none -t "${version}"-aks ./cmd/controller)"
+
+  img_repo="$(echo "${img}" | cut -d "@" -f 1 | cut -d ":" -f 1)"
+  img_tag="$(echo "${img}" | cut -d "@" -f 1 | cut -d ":" -f 2 -s)"
+  img_digest="$(echo "${img}" | cut -d "@" -f 2)"
+
+  img_registry="$(echo "${img_repo}" | cut -d "/" -f 1)"
+  img_path="$(echo "${img_repo}" | cut -d "/" -f 2-)"
```

Comment:
Same comment on, "could possibly use some minor comments on data format, and what exactly is getting extracted here. Little hard to just directly read/understand."

Reply:
Added example, hopefully helps

Reply:
This is great! Just having some info on the data format is perfect!